### DISCUSSION TITLE
ASGS Interface

### DISCUSSION THREAD

**Discussion by jasonfleming** [StormSurgeLive/asgs#1127] [General]
In general, the interface between ASGS and data consumers has always been ad-hoc, for better or worse, with metadata being embedded in the

1. thredds/nginx url path
2. ASGS instance name
3. email subject line for results notification
4. netcdf global attributes and variable-level attributes
5. ASGS instance status (`asgs.instance.status.json`) files
6. initialization/nowcast/forecast cycle status (`hook.status.json`) files
7. scenario status (`scenario.status.json`) files
8. scenario metadata (`run.properties`) files

The [ASGS Interface Guide](https://github.com/StormSurgeLive/asgs/wiki/ASGS-Interface-Guide) wiki page is a nice start but it is incomplete and out of date.

Is it possible to rationalize all these sources and channels of metadata (ref #1123)?

**Comment by wwlwpd** [comment]
When you have some time, can you just the first 1 or 2 and explain what you mean?

**Comment by jasonfleming** [comment]
Yes 
1. the url path contains year/basinstormnumber/advisory/mesh/host/instancename/scenario
2. the instancename often contains the mesh, operator ID, machine name, meteorological forcing etc
